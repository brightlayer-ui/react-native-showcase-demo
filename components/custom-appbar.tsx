import { blueDarkAlt } from '@pxblue/react-native-themes';
import React from 'react';
import { Appbar, useTheme } from 'react-native-paper';

export const MyCustomAppbar: typeof Appbar = (props) => {
    const theme = useTheme(props.theme);

    return <Appbar {...props} theme={theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}} />;
};
