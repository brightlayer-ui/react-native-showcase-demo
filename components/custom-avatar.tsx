import { blueDarkAlt } from '@pxblue/react-native-themes';
import React from 'react';
import { Avatar, useTheme } from 'react-native-paper';

export const MyCustomIconAvatar: typeof Avatar.Icon = (props) => {
    const theme = useTheme(props.theme);

    return <Avatar.Icon {...props} theme={theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}} />;
};

export const MyCustomImageAvatar: typeof Avatar.Image = (props) => {
    const theme = useTheme(props.theme);

    return <Avatar.Image {...props} theme={theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}} />;
};

export const MyCustomTextAvatar: typeof Avatar.Text = (props) => {
    const theme = useTheme(props.theme);

    return <Avatar.Text {...props} theme={theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}} />;
};
