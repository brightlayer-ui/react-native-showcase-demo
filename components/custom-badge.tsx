import { blueDarkAlt } from '@pxblue/react-native-themes';
import React from 'react';
import { Badge, useTheme } from 'react-native-paper';

export const MyCustomBadge: typeof Badge = (props) => {
    const theme = useTheme(props.theme);

    return (
        <Badge {...props} theme={theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}}>
            {props.children}
        </Badge>
    );
};
