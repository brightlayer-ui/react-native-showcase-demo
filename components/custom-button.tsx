import { blueDarkAlt } from '@pxblue/react-native-themes';
import React from 'react';
import { Button, useTheme } from 'react-native-paper';

export const MyCustomButton: typeof Button = (props) => {
    const theme = useTheme();
    return (
        <Button
            {...props}
            theme={props.mode === 'contained' && theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}}
        />
    );
};
