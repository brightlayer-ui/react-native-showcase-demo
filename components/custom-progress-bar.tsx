import { blueDarkAlt } from '@pxblue/react-native-themes';
import React from 'react';
import { ProgressBar, useTheme } from 'react-native-paper';

export const MyCustomProgressBar: typeof ProgressBar = (props) => {
    const theme = useTheme(props.theme);
    return <ProgressBar {...props} theme={theme.dark ? Object.assign({}, blueDarkAlt, props.theme) : {}} />;
};
