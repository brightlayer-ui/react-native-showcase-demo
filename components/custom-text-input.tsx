import { blueDarkAlt } from '@pxblue/react-native-themes';
import React from 'react';
import { TextInput, useTheme } from 'react-native-paper';

export const MyCustomTextInput: typeof TextInput = (props) => {
    const theme = useTheme(props.theme);
    const darkTheme = JSON.parse(JSON.stringify(blueDarkAlt));
    darkTheme.colors.primary = theme.colors.primary;

    return <TextInput {...props} theme={theme.dark ? Object.assign({}, darkTheme, props.theme) : {}} />;
};
