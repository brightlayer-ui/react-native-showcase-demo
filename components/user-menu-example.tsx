import { InfoListItem, InfoListItemProps, UserMenu, wrapIcon } from '@pxblue/react-native-components';
import React, { useState } from 'react';
import { View } from 'react-native';
import { Card, Divider } from 'react-native-paper';
import * as PXBColors from '@pxblue/colors';
import MatIcon from 'react-native-vector-icons/MaterialIcons';
const VpnKeyIcon = wrapIcon({ IconClass: MatIcon, name: 'vpn-key', flip: false });
const SettingsIcon = wrapIcon({ IconClass: MatIcon, name: 'settings', flip: false });
const ExitToAppIcon = wrapIcon({ IconClass: MatIcon, name: 'exit-to-app', flip: false });
const avatarTestImage = require('../assets/images/test-avatar.png');

export const UserMenuExample: React.FC = () => {
    const [menuClose, setMenuClose] = useState(false);

    const closeMenu = (): void => {
        setMenuClose(true);
        setTimeout((): void => setMenuClose(false), 0);
    };

    const customMenu = (): JSX.Element => (
        <View>
            <InfoListItem title={'Custom Menu Title'} subtitle={'Custom Menu Subtitle'} />
            <Divider />
            <InfoListItem title={'Custom Menu Info List Item 1'} onPress={(): void => closeMenu()} />
            <InfoListItem title={'Custom Menu Info List Item 2'} onPress={(): void => closeMenu()} />
        </View>
    );

    const menuItems: InfoListItemProps[] = [
        { title: 'Change Password', IconClass: VpnKeyIcon },
        { title: 'Preferences', IconClass: SettingsIcon },
        { title: 'Log Out', IconClass: ExitToAppIcon },
    ];

    return (
        <>
            <Card style={{ padding: 0, margin: 10, marginBottom: 0 }}>
                <View
                    style={{ flexDirection: 'row', justifyContent: 'center', marginHorizontal: 24, marginVertical: 24 }}
                >
                    <UserMenu
                        avatarIcon={'account-circle'}
                        avatarType={'icon'}
                        avatarColor={PXBColors.white[50]}
                        avatarSize={56}
                        avatarBackgroundColor={PXBColors.red[500]}
                        menu={customMenu()}
                        menuClose={menuClose}
                    />
                    <UserMenu
                        avatarImage={avatarTestImage}
                        avatarType={'image'}
                        avatarSize={56}
                        avatarBackgroundColor={PXBColors.green[500]}
                        menuTitle={'John Doe'}
                        menuSubtitle={'j.doe@example.com'}
                        menuItems={menuItems}
                    />
                    <UserMenu
                        avatarText={'PX'}
                        avatarType={'text'}
                        avatarColor={PXBColors.white[50]}
                        avatarSize={56}
                        avatarBackgroundColor={PXBColors.black[500]}
                        menuTitle={'John Doe'}
                        menuSubtitle={'j.doe@example.com'}
                        menuItems={menuItems}
                    />
                </View>
            </Card>
            <Card style={{ padding: 0, margin: 10, marginBottom: 0 }}>
                <View
                    style={{ flexDirection: 'row', justifyContent: 'center', marginHorizontal: 24, marginVertical: 24 }}
                >
                    <UserMenu
                        avatarIcon={'account-circle'}
                        avatarType={'icon'}
                        avatarColor={PXBColors.white[50]}
                        menu={customMenu()}
                        menuClose={menuClose}
                    />
                    <UserMenu
                        avatarImage={avatarTestImage}
                        avatarType={'image'}
                        menuTitle={'John Doe'}
                        menuSubtitle={'j.doe@example.com'}
                        menuItems={menuItems}
                    />
                    <UserMenu
                        avatarText={'PX'}
                        avatarType={'text'}
                        avatarColor={PXBColors.white[50]}
                        menuTitle={'John Doe'}
                        menuSubtitle={'j.doe@example.com'}
                        menuItems={menuItems}
                    />
                </View>
            </Card>
        </>
    );
};
