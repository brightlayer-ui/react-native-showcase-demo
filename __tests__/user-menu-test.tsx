import 'react-native';
import React from 'react';
import { UserMenuExample } from '../components/user-menu-example';
import renderer from 'react-test-renderer';
jest.useFakeTimers();

describe('Kitchen sink snapshot', () => {
    test('Renders correctly and snapShot matches', ()=>{
    const snap = renderer.create(
            <UserMenuExample onToggleRTL={() => {}} onToggleTheme={() => {}} />
    ).toJSON();
    expect(snap).toMatchSnapshot();
})
});