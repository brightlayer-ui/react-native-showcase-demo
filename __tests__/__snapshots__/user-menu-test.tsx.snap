// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kitchen sink snapshot Renders correctly and snapShot matches 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    color="#007bc1"
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#e0eff8",
          "borderRadius": 40,
          "height": 80,
          "width": 80,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {},
      ]
    }
    testID="avatar"
  >
    <Text
      accessibilityElementsHidden={true}
      allowFontScaling={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#007bc1",
            "fontSize": 48,
          },
          Array [
            Object {
              "lineHeight": 48,
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
              ],
            },
            Object {
              "backgroundColor": "transparent",
            },
          ],
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰀉
    </Text>
  </View>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      deviceHeight={null}
      deviceWidth={null}
      forwardedRef={[Function]}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      panResponderThreshold={4}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "margin": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 144,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "borderRadius": 80,
                  "height": 80,
                  "width": 80,
                },
                undefined,
                Object {
                  "marginLeft": 16,
                },
              ]
            }
          >
            <View
              color="#007bc1"
              style={
                Array [
                  Object {
                    "backgroundColor": "#e0eff8",
                    "borderRadius": 40,
                    "height": 80,
                    "width": 80,
                  },
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  Object {},
                ]
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#007bc1",
                      "fontSize": 48,
                    },
                    Array [
                      Object {
                        "lineHeight": 48,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰀉
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#424e54",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "OpenSans-SemiBold",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#424e54",
                    },
                    undefined,
                  ],
                ]
              }
            >
              John Smith
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#424e54",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {},
                    Object {
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    Object {},
                    undefined,
                    Object {
                      "color": "#424e54",
                    },
                  ]
                }
              >
                Account Manager
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              Object {
                "marginLeft": -0,
                "marginRight": -0,
              },
            ]
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 104,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 80,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 80,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={true}
                style={
                  Array [
                    Object {
                      "color": "#727e84",
                      "fontSize": 24,
                    },
                    Object {},
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#424e54",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "OpenSans-SemiBold",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#424e54",
                    },
                    Object {
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ],
                ]
              }
            >
              Toggle RTL
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 104,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 80,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 80,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={true}
                style={
                  Array [
                    Object {
                      "color": "#727e84",
                      "fontSize": 24,
                    },
                    Object {},
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#424e54",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "OpenSans-SemiBold",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#424e54",
                    },
                    Object {
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ],
                ]
              }
            >
              Toggle Theme
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "minHeight": 104,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 80,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                    "width": 80,
                  },
                  undefined,
                ]
              }
            >
              <Text
                allowFontScaling={true}
                style={
                  Array [
                    Object {
                      "color": "#727e84",
                      "fontSize": 24,
                    },
                    Object {},
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#424e54",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "OpenSans-SemiBold",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "letterSpacing": 0,
                  },
                  Object {},
                  undefined,
                  Array [
                    Object {
                      "color": "#424e54",
                    },
                    Object {
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ],
                ]
              }
            >
              Cancel
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </Modal>
</View>
`;
